src/v2/Apps/Artists/Routes/ArtistsByLetter.tsx

```diff
@@ -52,19 +55,24 @@ export const ArtistsByLetter: React.FC<ArtistsByLetterProps> = ({
     },
   } = viewer
 
-  const handleNext = () => {
-    handleClick(endCursor)
+  const handleNext = (page: number) => {
+    handleClick(endCursor, page)
   }
 
-  const handleClick = (cursor: string) => {
+  const handleClick = (cursor: string, page: number) => {
     setLoading(true)
 
-    relay.refetch({ after: cursor }, null, error => {
+    relay.refetch({ page }, null, error => {
       if (error) {
         console.error(error)
       }
 
       setLoading(false)
+
+      router.push({
+        pathname: location.pathname,
+        query: { ...location.query, page },
+      })
```

Comment:
We can alternatively do this without the `relay.refetch`, which would give us the purple bar loader. Doing it manually lets us just fade out the links and replace them. There's only the single fetch either way because of the normalized cache, and the back button works fine. Kind of interesting to see this work as it does.